state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,262 @@
+                       " to cancel: ")
+
+    # if user opts to cancel the test, return None, handle in the next function
+    if start_test.lower() != "y":
+        return None, None, None
+    print("Collecting IMU data...")
+
+    # collect acceleration and gyroscope readings, return None mag default
+    time.sleep(wait_time)
+    acc = cubesat.acceleration()
+    gyro = cubesat.gyro()
+    print("Data Collection Complete")
+    return acc, gyro, None
+
+
+def stationary_imu_test(result_dict):
+    """
+    All stationary test automation happens in this function
+    Process user test results and update the result dictionary accordingly
+    """
+    # record acceleration
+    acc, gyro, mag = stationary_user_test()
+    if acc is None and gyro is None:
+        # user entered "n", cancel the test
+        result_dict['IMU_AccStationary'] = (
+            "Stationary test not completed.", False)
+        result_dict['IMU_GyroStationary'] = (
+            "Stationary test not completed.", False)
+        return result_dict
+
+    # else continue running the test
+    result_val_string_acc = f"Acc: {acc}"
+    result_val_string_gyro = f"Gyro: {gyro}"
+
+    # if total acceleration ~= 9.8 m/s^2 and gyro ~= 0 deg/s, true
+    result_val_bool_acc = 9.8 - norm(acc) < 0.2
+    result_val_bool_gyro = norm(gyro) < 0.2
+    result_dict['IMU_AccStationary'] = (
+        result_val_string_acc, result_val_bool_acc)
+    result_dict['IMU_GyroStationary'] = (
+        result_val_string_gyro, result_val_bool_gyro)
+    return result_dict
+
+
+def moving_user_test():
+    """
+    All user interaction for the moving test happens in this function
+    Set wait times, prompt the users and gather IMU data and input
+    """
+    # print user prompts
+    start_test = input("Type Y to start the moving test, any key to cancel: ")
+
+    # if user opts to cancel the test, return None, handle in the next function
+    if start_test.lower() != "y":
+        return None, None, None
+
+    # collect acc and gyro readings, return None mag
+    print(f"Please move the cubesat around for the next {wait_time} " +
+          "seconds, starting now.")
+    print("Collecting IMU data...")
+    time.sleep(wait_time // 2)
+    acc = cubesat.acceleration()
+    gyro = cubesat.gyro()
+    time.sleep(wait_time - wait_time // 2)
+    print("Data Collection Complete")
+    return acc, gyro, None
+
+
+def moving_imu_test(result_dict):
+    """
+    All moving test automation happens in this function
+    Process user test results and update the result dictionary accordingly
+    """
+    # record acceleration
+    acc, gyro, mag = moving_user_test()
+
+    if acc is None:
+        # user entered "n", cancel the test
+        result_dict['IMU_AccMoving'] = ("Moving test not completed.", False)
+        return result_dict
+
+    # else continue running the test
+    result_val_string = (f"Acc: {acc}")
+    result_val_bool = norm(acc) >= 0.1
```

Comment:
`result_val_bool` isn't very descriptive, good we call this something like `imu_is_moving`  or something.
Also this will be true almost certainly as when a imu is stationary it should have a norm of about 9.8.

Reply:
Instead of this I recommend checking something like over the span of two seconds, the average norm of the change in acceleration is big enough. By norm of the change in acceleration I mean:
```python
Δa = norm(a_prev-a_cur)
```

Reply:
I opted (based on your comment about doing 10 reads/second) to just calculate the average readings and check the norm of those.

Reply:
Sounds good